ci-operator/step-registry/configure-registry-storage/deploy-s3-cloudfront/configure-registry-storage-deploy-s3-cloudfront-commands.sh

```diff
@@ -0,0 +1,159 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    echo "Setting proxy"
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+AWSCRED="${CLUSTER_PROFILE_DIR}/.awscred"
+if [[ -f "${AWSCRED}" ]]; then
+  AWS_ACCESS_KEY_ID=$(cat "${AWSCRED}" | grep aws_access_key_id | tr -d ' ' | cut -d '=' -f 2) && export AWS_ACCESS_KEY_ID
+  AWS_SECRET_ACCESS_KEY=$(cat "${AWSCRED}" | grep aws_secret_access_key | tr -d ' ' | cut -d '=' -f 2) && export AWS_SECRET_ACCESS_KEY
+else
+  echo "Did not find compatible cloud provider cluster_profile"
+  exit 1
+fi
+
+# get region to deploy new bucket
+AWS_REGION=$(oc get -o jsonpath='{.status.platformStatus.aws.region}' infrastructure cluster) && export AWS_REGION
+NEW_BUCKET=$(head /dev/urandom | tr -dc a-z | head -c 10) && export NEW_BUCKET
+
+# create file to deploy cloudfront using new bucket
+cat >>"${SHARED_DIR}/create_aws_bucket.tf" <<EOF
+provider "aws" {
+  region = "${AWS_REGION}"
+  access_key = "${AWS_ACCESS_KEY_ID}"
+  secret_key = "${AWS_SECRET_ACCESS_KEY}"
+}
+
+resource "aws_s3_bucket" "ir_s3" {
+  bucket = "${NEW_BUCKET}"
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_acl" "ir_bucket_acl" {
+  bucket = aws_s3_bucket.ir_s3.id
+  acl    = "private"
+}
+
+resource "aws_s3_bucket_policy" "ir_s3_policy" {
+  bucket = aws_s3_bucket.ir_s3.id
+  policy = data.aws_iam_policy_document.ir_s3_policy_test.json
+}
+
+data "aws_iam_policy_document" "ir_s3_policy_test" {
+  statement {
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+
+    actions = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:DeleteObject",
+    ]
+
+    resources = [
+      "arn:aws:s3:::${NEW_BUCKET}/*",
+    ]
+  }
+}
+
+locals {
+  s3_origin_id = "K3FZYAKDQYAWWP"
```

Comment:
what's this id?

Reply:
It's a public key id I have added in the cloudfront setting home page https://us-east-1.console.aws.amazon.com/cloudfront/v3/home?region=us-east-1#/publickey